### ISSUE TITLE
A meta-package to install all UCSC kent tools?

### ISSUE THREAD

**Issue by maximilianh** [bioconda/bioconda-recipes#43618]
Hi @dpryan79, various people e.g. @diekhans and @mrvollger have asked if there was a conda package to install all the kent tools. I don't know how to figure that out. Maybe there is already a meta-package that pulls in all the kent-ucsc-tools? Or a command that I am missing? Thanks!

**Comment by mrvollger** [comment]
So I looked into it, and it is too large (~3.8GB) to install all at once, but I think we could have larger sets of related utilities that could be jointly installed. For example, all bed utilities and all bigBed utilities. 

I made a PR that does this here:
https://github.com/bioconda/bioconda-recipes/pull/43664

**Comment by maximilianh** [comment]
Hi Mitchell, hm, is 4GB really too large these days? Interesting. My watch
has almost 10 times more... :-)

Great to see the PR! Would you also want to add the tools hubCheck and the
bigWig* tools? We have lists of tools by "importance" and the group may
come up with somewhat more extended lists, e.g. anything important for
making track hubs: faToTwoBit, bptForTwoBit, hubCheck, bedGraphToBigWig,
wigToBigWig, fetchChromSizes, bigGuessDb, pslToBigPsl.

I personally use featureBits and overlapSelect a lot and for alignments, in
addition to blat, a few other tools, e.g. pslReps and pslCDnaFilter,
pslSort, pslMap, pslUniq, pslToChain, chainToAxt, chainToPsl, liftUp,
pslPretty, pslPairs.

**Comment by diekhans** [comment]
Maybe the limit could be appealed?

**Comment by mrvollger** [comment]
@dpryan79 and @johanneskoester, is it possible to have the resource limit increased so we could make this one package? It would be really useful, I think?

**Comment by bgruening** [comment]
I don't think this is possible at the moment. Its a lot of data if every CI job wants to allocate that.
We could try disabling the container tests and builds, that will reduce storage needs ... maybe I'm just building this meta-package locally for you. Its a one-time thing, isn't it? That always pull the lastest kent tools?

**Comment by peterjc** [comment]
How many misc tools are there outside of all bed utilities and all bigBed utilities?

e.g. isPcr

**Comment by mrvollger** [comment]
I think there are a little over 300 tools in the Kent/ucsc utils package.

**Comment by maximilianh** [comment]
I think most of them get very little use and I bet that we can cover 90% of
uses with around 30-40 tools

**Comment by mrvollger** [comment]
I think that is true. However, if we don't bundle them in some way then we might need 100s of different bioconda PRs everytime there is an update from your team (UCSC). 

I am not against making a couple of PRs to split the popular tools and then all the others or some combination.

**Comment by maximilianh** [comment]
>
> I think that is true. However, if we don't bundle them in some way then we
> might need 100s of different bioconda PRs everytime there is an update from
> your team (UCSC).
>

Sorry don't know what you mean. Isn't this ticket about bundling them?


> I am not against making a couple of PRs to split the popular tools and
> then all the others or some combination
>
Let me reformulate this: I'm pretty sure that there are a lot of tools that
are used by 0 pipelines out there.

But maybe it's easier just to increase the limit for once for this package
and waste some storage and break a rule, but save a lot of time?
